.github/workflows/test_models_go_tests.yml

```
@@ -0,0 +1,96 @@
+name: Library Go tests
+
+on:
+  workflow_call:
+    inputs:
+      dafny:
+        description: "The Dafny version to run"
+        required: true
+        type: string
+      num_shards:
+        required: false
+        type: number
+        default: 5
+
+jobs:
+  populate-matrix-dimensions:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Populate shard list
+        id: populate-shard-list
+        run: echo "shard-list=[`seq -s , 1 ${{ inputs.num_shards }}`]" >> $GITHUB_OUTPUT
+    outputs:
+      shard-list: ${{ steps.populate-shard-list.outputs.shard-list }}
+  testGo:
+    needs: populate-matrix-dimensions
+    strategy:
+      fail-fast: false # at least for development; see all errors
+      matrix:
+        shard: ${{ fromJson(needs.populate-matrix-dimensions.outputs.shard-list) }}
+    runs-on: "ubuntu-latest"
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      DOTNET_CLI_TELEMETRY_OPTOUT: 1
+      DOTNET_NOLOGO: 1
+    steps:
+      - name: Support longpaths on Git checkout
+        run: |
+          git config --global core.longpaths true
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::370957321024:role/GitHub-CI-PolymorphTestModels-Role-us-west-2
+          role-session-name: PythonTests
+
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      # TODO: Use the released version once all the Dafny issues aew fixed.
+      - name: Setup Dafny
+        uses: dafny-lang/setup-dafny-action@v1.7.0
+        with:
+          dafny-version: "nightly-latest"
```

```suggestion
          dafny-version: "nightly-2024-12-04-bf1aac6"
```
Locking down a fixed prerelease for determinisitic CI. The nightly build will take care of catching regressions instead.